include/eve/module/core/regular/impl/simd/x86/rsqrt.hpp

```diff
@@ -32,151 +32,156 @@ namespace eve::detail
+    a0 = c8 * a0 * fnma(y, fnma(c3, y, c10), c15);
+
+    if constexpr( std::is_same_v<v_t, double> )
+    {
+      // To obtain extra accuracy, we need one additional Newton step
+      a0 = fma(fnma(x, sqr(a0), one(eve::as(a0))), a0 * half(eve::as(a0)), a0);
+    }
+
+    if constexpr( platform::supports_infinites )
+    {
+      a0 = if_else(x == inf(eve::as(x)), eve::zero, a0);
+    }
+
+    return if_else(is_eqz(x), inf(eve::as(x)), a0);
   }
 
-  if constexpr( platform::supports_infinites )
+  template<typename Pack> EVE_FORCEINLINE Pack
+  rsqrt_x86_full(Pack const& x) noexcept
   {
-    a0 = if_else(x == inf(eve::as(x)), eve::zero, a0);
+    using v_t = typename Pack::value_type;
+    if( eve::any(is_denormal(x)) ||
+        (std::is_same_v<v_t, double> && eve::any(eve::abs(x) < smallestposval(eve::as<float>()) ||
+                                                 eve::abs(x) > valmax(eve::as<float>()))) )
+      // this is necessary because of the poor initialisation by float intrinsic
+    {
+      auto [a00, nn] = ifrexp[pedantic](x);
+      auto tst       = is_odd(nn);
+      nn             = dec[tst](nn);
+      a00            = mul[tst](a00, 2);
+      auto a0        = rsqrt_x86_normal(a00);
+      std::cout << ldexp[pedantic](a0, -nn / 2) << std::endl;
+      std::cout << is_eqz(x) << std::endl;
+      return if_else(is_eqz(x), inf(eve::as(x)), ldexp[pedantic](a0, -nn / 2));
+    }
+    else
+    {
+      return rsqrt_x86_normal(x);
+    }
   }
 
-  return if_else(is_eqz(x), inf(eve::as(x)), a0);
-}
-
-template<typename Pack>
-EVE_FORCEINLINE Pack
-rsqrt_x86_pedantic(Pack const& x) noexcept
-{
-  using v_t = typename Pack::value_type;
-  if( eve::any(is_denormal(x))
-      || (std::is_same_v<
-              v_t,
-              double> && eve::any(eve::abs(x) < smallestposval(eve::as<float>()) || eve::abs(x) > valmax(eve::as<float>()))) )
-  // this is necessary because of the poor initialisation by float intrinsic
+//------------------------------------------------------------------------------------------------
+//128 bits rsqrt
+  template<floating_scalar_value T, typename N, callable_options O>
+  EVE_FORCEINLINE wide<T, N> rsqrt_(EVE_REQUIRES(sse2_),
+                                    O          const&,
+                                    wide<T, N> const& a0) noexcept
+  requires std::same_as<abi_t<T, N>, x86_128_>
   {
-    auto [a00, nn] = ifrexp[pedantic](x);
-    auto tst       = is_odd(nn);
-    nn             = dec[tst](nn);
-    a00            = mul[tst](a00, 2);
-    auto a0        = rsqrt_x86(a00);
-    return if_else(is_eqz(x), inf(eve::as(x)), ldexp[pedantic](a0, -nn / 2));
+    if constexpr(O::contains(raw2))
+    {
+      if constexpr( std::is_same_v<T, double> )
+      {
+        // The maximum error for this approximation is 1.5e-12
+        return _mm_cvtps_pd(_mm_rsqrt_ps(_mm_cvtpd_ps(a0)));
+      }
+      else if constexpr( std::is_same_v<T, float> )
+      { return _mm_rsqrt_ps(a0); }
```

Comment:
skip lines or remove the { }  and put on previosu line

Reply:
done